lib/Conversion/TTIRToTTNN/TTIRToTTNN.cpp

```diff
@@ -542,6 +542,43 @@ class MaxPool2dOpConversionPattern
   }
 };
 
+class BroadcastOpConversionPattern
+    : public OpConversionPattern<ttir::BroadcastOp> {
+  using OpConversionPattern<ttir::BroadcastOp>::OpConversionPattern;
+
+public:
+  LogicalResult
+  matchAndRewrite(ttir::BroadcastOp srcOp, ttir::BroadcastOp::Adaptor adaptor,
+                  ConversionPatternRewriter &rewriter) const override {
+
+    mlir::Value input = srcOp.getOperand(0);
+    mlir::Value result = srcOp.getResult();
+
+    SmallVector<Operation *> srcOpUsers;
+    for (Operation *nextOp : srcOp->getUsers()) {
+      srcOpUsers.push_back(nextOp);
+    }
+
+    // Try to fold this operation into all Element wise consumer ops.
+
+    for (Operation *nextOp : srcOpUsers) {
+      StringRef opName = nextOp->getName().getStringRef();
+      if (not opName.starts_with("ttir.broadcast") and
```

Comment:
It would be better to use type comparison with `mlir::isa` instead of name string comparison

Reply:
OK but this part is now removed.